Update and clarify preview instructions.

mbrukman opened the pull request:
Provide more details and clarity about how to submit changes for this
repo and how to make a preview of the website to enable sharing with
reviewers and other observers.

hsaputra commented on README.md:
````diff
@@ -42,30 +42,92 @@ Before pushing the site to production, and a great way to demonstrate proposed
+
+   ```
+   $ git checkout master
+   $ git pull upstream master
````
Assuming my clone is old one, we should also do `git merge upstream/master` to make sure local `master` has latest from the upstream repo.

hsaputra commented:
As far as I know doing `git pull upstream master` will only update your
local upstream/master copy.

Need to merge to your fork of origin/master to sync with upstream:

https://help.github.com/articles/syncing-a-fork/

Thanks,

- Henry


On Sun, Apr 22, 2018, 1:18 PM Misha Brukman <notifications@github.com>
wrote:

> *@mbrukman* commented on this pull request.
> ------------------------------
>
> In README.md
> <https://github.com/JanusGraph/janusgraph.org/pull/41#discussion_r183249908>
> :
>
> > -1. visit the "Settings" page for your fork on GitHub and make the `gh-pages`
> +
> +1. Clone your copy of the repo locally
> +
> +   ```
> +   $ git clone git@github.com:[USERNAME]/janusgraph.org.git
> +   $ cd janusgraph.org
> +   $ git remote add upstream git@github.com:JanusGraph/janusgraph.org.git
> +   ```
> +
> +1. Create a new branch for your changes, based on the latest changes in the
> +   `master` branch of the upstream repo:
> +
> +   ```
> +   $ git checkout master
> +   $ git pull upstream master
>
> git pull upstream master will do exactly that, no?
>
> What I typically run myself is actually:
>
> git pull upstream master --ff-only
>
> which ensures a fast-forward (clean) merge, and it fails if I have any
> changes on my master added accidentally, which should have gone through a
> feature branch and merged upstream first, before showing up on my local
> master, which I try to keep clean and inline with upstream.
>
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub
> <https://github.com/JanusGraph/janusgraph.org/pull/41#discussion_r183249908>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAGb2poA58mWHapYhHlNhiXZ5XbdF8tNks5trOWVgaJpZM4Te9cJ>
> .
>

mbrukman commented:
@hsaputra — what I typically do is:

Bring local `master` to match upstream:

```
$ git checkout master
$ git pull upstream master
```

If I want my GitHub fork's `master` to be up-to-date (which is really optional, but why not):

```
$ git push    # automatically pushes local `master` to `origin/master`
```

If I want to update my feature branch:

```
$ git checkout my-feature-branch
$ git rebase master
$ git push -f     # update my PR or in-progress branch to have a clean diff
```

If I merge `master` into my feature branch, it becomes more complex and has a lot of extra spurious commits, whereas I want it to only have a single commit (or 2 if that's what I'm trying to do specifically).

The rebasing of my feature branch on the updated `master` is described lower in the README.

Does that address your concern, or were you looking for something else?

hsaputra commented:
Ohhh yes, you are right. That will work too.

I miss where it is checkout the origin master branch.

Will update my review, thx

- Henry


On Sun, Apr 22, 2018, 2:44 PM Misha Brukman <notifications@github.com>
wrote:

> @hsaputra <https://github.com/hsaputra> — what I typically do is:
>
> Bring local master to match upstream:
>
> $ git checkout master
> $ git pull upstream master
>
> If I want my GitHub fork's master to be up-to-date (which is really
> optional, but why not):
>
> $ git push    # automatically pushes local `master` to `origin/master`
>
> If I want to update my feature branch:
>
> $ git checkout my-feature-branch
> $ git rebase master
> $ git push -f     # update my PR or in-progress branch to have a clean diff
>
> If I merge master into my feature branch, it becomes more complex and has
> a lot of extra spurious commits, whereas I want it to only have a single
> commit (or 2 if that's what I'm trying to do specifically).
>
> The rebasing of my feature branch on the updated master is described
> lower in the README.
>
> Does that address your concern, or were you looking for something else?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/JanusGraph/janusgraph.org/pull/41#issuecomment-383414776>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAGb2ov7xwaiUYOrkuUDEKtuEn-FXHNvks5trPnDgaJpZM4Te9cJ>
> .
>

hsaputra commented on README.md:
````diff
@@ -42,30 +42,93 @@ Before pushing the site to production, and a great way to demonstrate proposed
+   ```
+   $ git clone git@github.com:[USERNAME]/janusgraph.org.git
+   $ cd janusgraph.org
+   $ git remote add upstream git@github.com:JanusGraph/janusgraph.org.git
````
Dont you need to do `git fetch upstream` right after this?

hsaputra replied:
Never mind, will be pulled later.

hsaputra commented:
@mbrukman Added one last question/ comment on the PR.

mbrukman commented:
@hsaputra — I also remembered and added the use of `--ff-only` flag to ensure that `master` is a clean copy of upstream and added some documentation about this. Please take a look and let me know what you think. Thanks for the review!

mbrukman commented:
@hsaputra — is the PR still OK to merge? See my most recent comment above for changes made since your last review. Let me know if you have any concerns before I merge. Thanks!

hsaputra commented:
@mbrukman Yeah, LGTM

Sorry I didn't make it clear.

mbrukman commented:
Thanks, @hsaputra and @amcp, for the reviews!
